bin/build_config.py

```diff
@@ -1,87 +1,296 @@
-        'genes.json',
-    ],
-    'neighborhoods': [
-        'neighborhoods.json'
-    ],
-    'genomic-profiles': [
-        'genomic-profiles.zarr'
-    ],
-}
-
-def write_raster_json(
-    title='',
-    dataset='',
-    files_dir='',
-    zarr_dirs=None,
-    url='',
-    outdir='',
-    config_filename='config.json',
-    ):
-    config_json = {
-        "name": title,
-        "version": "1.0.0",
-        "description": "",
-        "public": True,
-        "datasets": [],
-        "initStrategy": "auto",
-        "coordinationSpace": {},
-        "layout": []
-    }
-
-    files = []
     for data_type in DATA_TYPES:
-        for file_type in DATA_TYPES[data_type]:
-            f = os.path.join(files_dir, file_type)
-            if os.path.isfile(f):
-                files.append({
-                    "url": file_type,
-                    "type": data_type,
-                    "fileType": file_type
-                })
-
-    # TODO: add zarr data
-    if zarr_dirs and type(zarr_dirs) != bool:
-        if type(zarr_dirs) not in [list, tuple]:
-            zarr_dirs = [zarr_dirs]
-    
-    config_json["datasets"].append({
-        "uid": dataset,
-        "name": dataset,
-        "description": "",
-        "files": files,
-    })
+        for file_name, file_type in DATA_TYPES[data_type]:
+
+            # first file type found will be used in the config file
+            file_exists = False
+            if file_name in file_paths_names:
+                file_path = file_paths_names[file_name]
+                file_exists = True
+
+            if file_exists:
+                has_files = True
+                if file_type in DEFAULT_OPTIONS:
+                    file_options = build_options(
+                        file_type, file_path, options or DEFAULT_OPTIONS[file_type]
+                    )
+                    # Set a coordination scope for any 'mapping'
+                    if "mappings" in file_options:
+                        coordination_types[ct.EMBEDDING_TYPE] = cycle(
+                            chain(
+                                coordination_types[ct.EMBEDDING_TYPE],
+                                [
+                                    config.set_coordination_value(
+                                        ct.EMBEDDING_TYPE.value, k, k
+                                    )
+                                    for k in file_options["mappings"]
+                                ],
+                            )
+                        )
+                else:
+                    file_options = None
+                config_dataset.add_file(
+                    data_type,
+                    file_type,
+                    url=os.path.join(url, os.path.basename(file_path)),
+                    options=file_options,
+                )
+                dts.add(data_type)
+                break
+
+    if not has_files:
+        logging.error("No files to add to config file")
+        quit(1)
```

Comment:
**_Anecdotal comment:_**
Philosofically speaking, different errors should raise different exit codes so `quit(1)`, `quit(2)`, etc. to represent different things going bad. An alternative is to `raise SystemExit("The error")` if you want a decriptive message too instead of going fo the exit codes and saves you the extra log.error(...)

Reply:
Thanks for the tip. Realized you lose the stack trace with `raise SystemExit` so I've kept a `logging.error` in one case, but I've removed all `quit(1)` 432963e